CONTRIBUTING.md

```diff
@@ -0,0 +1,724 @@
+amounts of code to the project.
+Rather the decision to on-board a member to an admin position requires a history
+of using and contributing to BioJulia, and a positive
+interaction and involvement with the community. Any BioJulia member fulfilling
+this, may offer to take on this [responsibility](#biojulia-administrators).
+
+## Styleguides
+
+### Git Commit messages
+
+* Use the present tense ("Add feature" not "Added feature").
+* Use the imperative mood ("Move cursor to..." not "Moves cursor to...").
+* Limit the first line to 72 characters or less.
+* Reference issues and pull requests liberally after the first line.
+* Consider starting the commit message with an applicable emoji:
+    * :art: `:art:` when improving the format/structure of the code
+    * :racehorse: `:racehorse:` when improving performance
+    * :memo: `:memo:` when writing docs
+    * :penguin: `:penguin:` when fixing something on Linux
+    * :apple: `:apple:` when fixing something on macOS
+    * :checkered_flag: `:checkered_flag:` when fixing something on Windows
+    * :bug: `:bug:` when fixing a bug
+    * :fire: `:fire:` when removing code or files
+    * :green_heart: `:green_heart:` when fixing the CI build
+    * :white_check_mark: `:white_check_mark:` when adding tests
+    * :arrow_up: `:arrow_up:` when upgrading dependencies
+    * :arrow_down: `:arrow_down:` when downgrading dependencies
+    * :exclamation: `:exclamation:` when removing warnings or depreciations
+
+### Additional julia style suggestions
+
+* Source code files should have the following style of header:
+
+  ```julia
+  # Title
+  # =====
+  #
+  # Short description.
+  #
+  # [Long description (optional)]
+  #
+  # This file is a part of BioJulia. License is MIT: <link to the license file>
+  ```
+
+* Indent with 4 spaces.
+
+* When returning values from a function, use an explicit `return` statement.
+Be aware that functions in julia implicitly return the the result of the last
+statement in the function.
+
+* Type names are camel case, with the first letter capitalized. E.g.
+  `SomeVeryUsefulType`.
+
+* Module names are also camel case.
+
+* Function names, apart from constructors, are all lowercase.
+  Include underscores between words only if the name would be hard
+  to read without.
+  E.g.  `start`, `stop`, `findletter` `find_last_digit`.
+
+* Generally try to keep lines below 80-columns, unless splitting a long line
+  onto multiple lines makes it harder to read.
+
+* Files that declare modules should only declare the module, and import any
+  modules that it requires. Any subsequent significant code should be included
+  from separate files. E.g.
+
+```julia
+module AwesomeFeatures
+
+using IntervalsTrees, JSON
+
+include("feature1.jl")
+include("feature2.jl")
+
+end
+```
+
+* Files that declare modules should have the same name name of the module.
+  E.g the module `SomeModule` is declared under the file `SomeModule.jl`.
+
+* Separate logical blocks of code with one blank line,
```

Comment:
This is another case, where typically when there are a number of related single line methods, current practice is to not have any blank lines between them, let alone 2.